core/aws-core/src/main/java/software/amazon/awssdk/awscore/internal/client/config/AwsClientOptionValidation.java

```diff
@@ -39,8 +39,8 @@ public static void validateSyncClientOptions(SdkClientConfiguration c) {
     }
 
     private static void validateClientOptions(SdkClientConfiguration c) {
-        require("credentialsProvider", c.option(AwsClientOption.CREDENTIALS_PROVIDER));
-
+        // TODO: where does the field name come from? Should it be "credentialsIdentityProvider"?
+        require("credentialsProvider", c.option(AwsClientOption.CREDENTIALS_IDENTITY_PROVIDER));
```

Comment:
This is just for helpful error messages. I think `credentialsProvider` is correct, since that's the setter we have on the AWS client.